### PR TITLE
Documentation for debugging CRD-based Zenith

### PR DESCRIPTION
**Pull request by mkjpryor**

### CODE REVIEW

File: docs/debugging/zenith-services.md [hunk 1]
```diff
@@ -20,29 +20,74 @@ persist, try restarting the Zenith SSHD:
 kubectl -n azimuth rollout restart deployment/zenith-server-sshd
 ```
 
-## Client not registered in Consul
+## Client not appearing in Zenith CRDs
 
-Once a client has connected to SSHD successfully, it should get registered in
-[Consul](https://www.consul.io/).
+The components of Zenith communicate using three Kubernetes CRDs:
```
Comment (m-bull) [inline docs/debugging/zenith-services.md:25]:
```suggestion
The components of Zenith communicate using three [Kubernetes CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/):
```

File: docs/debugging/zenith-services.md [hunk 2]
```diff
@@ -20,29 +20,74 @@ persist, try restarting the Zenith SSHD:
...
 
-To determine if this is the case, it is useful to access the Consul UI. As discussed
-in [Monitoring and alerting](../configuration/14-monitoring.md), the Consul UI
-is exposed as `consul.<ingress base domain>`, e.g. `consul.azimuth.example.org`,
-and is protected by a username and password.
+  * `services.zenith.stackhpc.com`  
+    Respresents a reserved domain and stores the associated public key.
```
Comment (m-bull) [inline docs/debugging/zenith-services.md:28]:
```suggestion
    A reserved domain and associated SSH public key.
```

File: docs/debugging/zenith-services.md [hunk 3]
```diff
@@ -20,29 +20,74 @@ persist, try restarting the Zenith SSHD:
...
+  * `endpoints.zenith.stackhpc.com`  
+    Stores the current endpoints for a Zenith service.
```
Comment (m-bull) [inline docs/debugging/zenith-services.md:30]:
```suggestion
    The current endpoints for a Zenith service.  
```

File: docs/debugging/zenith-services.md [hunk 4]
```diff
@@ -20,29 +20,74 @@ persist, try restarting the Zenith SSHD:
...
 
-To determine if this is the case, it is useful to access the Consul UI. As discussed
-in [Monitoring and alerting](../configuration/14-monitoring.md), the Consul UI
-is exposed as `consul.<ingress base domain>`, e.g. `consul.azimuth.example.org`,
-and is protected by a username and password.
+  * `services.zenith.stackhpc.com`  
+    Respresents a reserved domain and stores the associated public key.
+  * `endpoints.zenith.stackhpc.com`  
+    Stores the current endpoints for a Zenith service.  
+    Each SSH tunnel for a service updates this resource to add its address, port and configuration.
```
Comment (m-bull) [inline docs/debugging/zenith-services.md:31]: Does the SSH tunnel update this? Could be clearer in the mechanism I think.

File: docs/debugging/zenith-services.md [hunk 5]
```diff
@@ -20,29 +20,74 @@ persist, try restarting the Zenith SSHD:
...
+  * `leases.zenith.stackhpc.com`  
+    Stores heartbeat information for an individual SSH tunnel.
```
Comment (m-bull) [inline docs/debugging/zenith-services.md:33]:
```suggestion
    Heartbeat information for an individual SSH tunnel.  
```

File: docs/debugging/zenith-services.md (hunk 4 above)
Comment (m-bull) [inline docs/debugging/zenith-services.md:31]:
```suggestion
    Each Zenith SSH tunnel for a service updates this resource to add its address, port and configuration.
```

File: docs/debugging/zenith-services.md [hunk 6]
```diff
@@ -20,29 +20,74 @@ persist, try restarting the Zenith SSHD:
...
-is exposed as `consul.<ingress base domain>`, e.g. `consul.azimuth.example.org`,
-and is protected by a username and password.
+  * `services.zenith.stackhpc.com`  
+    Respresents a reserved domain and stores the associated public key.
+  * `endpoints.zenith.stackhpc.com`  
+    Stores the current endpoints for a Zenith service.  
+    Each SSH tunnel for a service updates this resource to add its address, port and configuration.
+  * `leases.zenith.stackhpc.com`  
+    Stores heartbeat information for an individual SSH tunnel.  
+    Each SSH tunnel has its own lease resource that is regularly updated with a heartbeat.
```
Comment (m-bull) [inline docs/debugging/zenith-services.md:34]:
```suggestion
    Each Zenith SSH tunnel has its own lease resource that is regularly updated with a heartbeat.
```

File: docs/debugging/zenith-services.md [hunk 7]
```diff
@@ -20,29 +20,74 @@ persist, try restarting the Zenith SSHD:
...
 
-The default view shows Consul's view of the services, where you can check if the
-service is being registered correctly.
+If a service is not functioning as expected, we should check the state of the CRDs for
```
Comment (m-bull) [inline docs/debugging/zenith-services.md:36]:
```suggestion
If a Zenith service is not functioning as expected, check the state of the CRDs for 
```

File: docs/debugging/zenith-services.md [hunk 8]
```diff
@@ -20,29 +20,74 @@ persist, try restarting the Zenith SSHD:
...
+    7tnqm:
+      address: 10.42.0.71
+      config:
+        backend-protocol: http
+        skip-auth: false
+      port: 42109
+      status: passing
+```
+
+The address should be the pod IP of an SSHD pod and the port should be the allocated port
```
Comment (m-bull) [inline docs/debugging/zenith-services.md:88]:
```suggestion
The address should be the pod IP of a Zenith SSHD pod, and the port should be the allocated port
```

File: docs/debugging/zenith-services.md [hunk 9]
```diff
@@ -67,45 +112,87 @@ the identity operator:
...
+```
+
 ## Kubernetes resources for the Zenith service have not been created
 
-If the service exists in Consul, it is possible that the process that synchronises Consul
-services with Kubernetes resources is not functioning correctly. To check if Kubernetes
-resources are being created, run the following command and check that the `Ingress`,
-`Service` and `Endpoints` resources have been created for the service:
+If the CRDs for the service look correct, it is possible that the component that watches
+the Zenith CRDs and hooks up the Kubernetes ingress for those services is not functioning
```
Comment (m-bull) [inline docs/debugging/zenith-services.md:124]:
```suggestion
the Zenith CRDs and creates the Kubernetes ingress for those services is not functioning
```

**Review by m-bull — CHANGES_REQUESTED** [review]

File: docs/debugging/zenith-services.md (hunk 4 above)
Comment (mkjpryor) [reply docs/debugging/zenith-services.md:31]: The script that runs at the server end of the tunnel does indeed update this.

File: docs/debugging/zenith-services.md [hunk 10]
```diff
@@ -20,29 +20,74 @@ persist, try restarting the Zenith SSHD:
...
 
-To determine if this is the case, it is useful to access the Consul UI. As discussed
-in [Monitoring and alerting](../configuration/14-monitoring.md), the Consul UI
-is exposed as `consul.<ingress base domain>`, e.g. `consul.azimuth.example.org`,
-and is protected by a username and password.
+  * `services.zenith.stackhpc.com`  
+    A reserved domain and associated SSH public key.
+  * `endpoints.zenith.stackhpc.com`  
+    The current endpoints for a Zenith service.  
+    Each Zenith SSH tunnel for a service updates this resource to add its address, port and configuration.
```
Comment (m-bull) [inline docs/debugging/zenith-services.md:31]:
```suggestion
    This resource is updated to add the address, port and configuration of the Zenith SSH tunnel as the SSH tunnel is created.
```

**Review by m-bull — APPROVED** [review]

**Pull request merged**